extensions/amazon-lambda/event-server/src/main/java/io/quarkus/amazon/lambda/runtime/MockEventServer.java

```diff
@@ -100,54 +95,56 @@ public void postEvent(RoutingContext ctx) {
     }
 
     public void nextEvent(RoutingContext ctx) {
-        // Vert.x barfs if you block too long so we have our own executor
-        blockingPool.execute(() -> {
-            final AtomicBoolean closed = new AtomicBoolean(false);
-            ctx.response().closeHandler((v) -> closed.set(true));
-            ctx.response().exceptionHandler((v) -> closed.set(true));
-            ctx.request().connection().closeHandler((v) -> closed.set(true));
-            ctx.request().connection().exceptionHandler((v) -> closed.set(true));
-            RoutingContext request = null;
-            try {
-                for (;;) {
-                    request = queue.poll(10, TimeUnit.MILLISECONDS);
-                    if (request != null) {
-                        if (closed.get()) {
-                            log.debugf("Polled message %s but connection was closed, returning to queue",
-                                    request.get(AmazonLambdaApi.LAMBDA_RUNTIME_AWS_REQUEST_ID));
-                            queue.put(request);
+        vertx.executeBlocking(new Handler<>() {
```

Comment:
Did you see my above comment?  Using the executeBlocking has a timeout and you may want to block FOR A LONG TIME.  Especially since we are in test mode.

Reply:
I did, but I misinterpreted it. No worries though, I'll update the PR tomorrow to set the timeout extremely high, thus preserving the essence of what this change attempts.

Reply:
PR updated.